pkg/transaction/eventgenerator.go

```diff
@@ -148,10 +148,12 @@ func (e *Generator) CreateEvents(summaryEvent LogEvent, detailEvents []LogEvent,
 	}
 
 	// Check to see if marketplace provisioning/subs is enabled
-	if agent.GetCentralClient().IsMarketplaceSubsEnabled() {
-		err := e.processTxnSummary(summaryEvent)
-		if err != nil {
-			return nil, err
+	if agent.GetCentralClient() != nil {
+		if agent.GetCentralClient().IsMarketplaceSubsEnabled() {
```

Comment:
this can be simplified

if agent.GetCentralClient() != nil && agent.GetCentralClient().IsMarketplaceSubsEnabled() {...}